lib/active_merchant/billing/gateways/worldpay.rb

```diff
@@ -441,15 +441,23 @@ def add_payment_method(xml, amount, payment_method, options)
       end
 
       def add_network_tokenization_card(xml, payment_method)
+        token_type = {
+          apple_pay: 'APPLEPAY',
+          google_pay: 'GOOGLEPAY',
+          network_token: 'NETWORKTOKEN'
+        }[payment_method.source]
+
         xml.paymentDetails do
-          xml.tag! 'EMVCO_TOKEN-SSL', 'type' => 'NETWORKTOKEN' do
+          xml.tag! 'EMVCO_TOKEN-SSL', 'type' => token_type do
             xml.tokenNumber payment_method.number
             xml.expiryDate do
               xml.date(
                 'month' => format(payment_method.month, :two_digits),
                 'year' => format(payment_method.year, :four_digits_year)
               )
             end
+            card_h_name = card_holder_name(payment_method, options)
```

Comment:
This mix of two words and one letter `card_h_name` is a little out of place; we usually avoid abbreviations. I assume you didn't want to mirror the method name, so in this case just using something like `name` would make sense.